pydash/strings.py

```diff
@@ -736,6 +736,56 @@ def lines(text):
     return text.splitlines()
 
 
+def lower_case(text):
+    """Converts string to lower case as space separated words. Before
+    converting the given text to lower case this method:
+
+      - replaces special character with white spaces,
+      - adds a white space before the capital letter,
+      - replaces multiple whitespaces with a single whitespace,
+      - pertains spaces between words in the given text,
+      - pertains the digits given in the text.
+
+    Args:
+        text (str): String to convert.
+
+    Returns:
+        str: String converted to lower case as space separated words.
+
+    Example:
+
+        >>> lower_case('fooBar')
+        'foo bar'
+        >>> lower_case('--foo-Bar--')
+        'foo bar'
+        >>> lower_case('/?*Foo10/;"B*Ar')
+        'foo10 b ar'
+    """
+    # TODO: Add release version.
+    characters = []
+
+    for char in text:
+        if not char.isalpha() and not char.isdigit():
+            # If any of the characters is neither an alphabet or a digit, then
+            # replace them with whitespaces.
+            characters.append(' ')
+        elif char.istitle():
+            # If any of the characters is a capital letter, then go ahead and
+            # add a whitespace before the capital letter.
+            characters.append(' {0}'.format(char))
+        else:
+            characters.append(char)
+
+    text = ''.join(characters)
+    text = text.strip()
+
+    # Replace multiple whitespaces between the words in the text with a single
+    # whitespace.
+    text = re.sub("\s\s+", " ", text)
+
+    return text.lower()
```

Comment:
BTW, I took a quick look at the Lodash regex for words and it's gotten quite a bit more complex and they now have two different regexes that are used for matching ASCII and Unicode separately (i.e. they check if the string has unicode or not and then use one of the two word regexes):

- reUnicodeWord: https://github.com/lodash/lodash/blob/da329eb776a15825c04ffea9fa75ae941ea524af/lodash.js#L257
- reAsciiWord: https://github.com/lodash/lodash/blob/da329eb776a15825c04ffea9fa75ae941ea524af/lodash.js#L161